archivist/dictmerge.py

```diff
@@ -28,3 +28,44 @@ def _dotdict(dct: Optional[dict]) -> Optional[dict]:
     if dct is None:
         return None
     return flatten(dct, reducer="dot")
+
+
+def assets_ext_attr(assets: list) -> list:
+    """Create list of assets with extended attribute(s)"""
+    extended_attributes_asset = [
+        asset
+        for asset in assets
+        if any(not item.startswith("arc_") for item in asset["attributes"])
+    ]
+    return extended_attributes_asset
+
+
+def attachment_identities_assets(assets_with_attachments: list) -> list:
+    """Create list of attachment identities"""
+    for asset in assets_with_attachments:
+        total_attachments_assets = {
+            item["arc_attachment_identity"]
+            for item in asset["attributes"]["arc_attachments"]
```

Comment:
arc_attachments may not exist so use the get() idiom instead.

            for item in asset["attributes"].get("arc_attachments", [])

Reply:
Change made